### PR TITLE
Simplify the node name for metrics - just use localhost

### PR DESCRIPTION
**Pull request by dims**
Simplest replacement for overly complicated https://github.com/kubernetes/kubernetes/pull/118932

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e_node/util.go [hunk 1]
```diff
@@ -641,20 +640,8 @@ func WaitForPodInitContainerToFail(ctx context.Context, c clientset.Interface, n
...
-			// yay, node name resolved correctly, pick the first
-			return ips[0].String()
-		}
-	}
-	// fallback to node name in test context
-	return framework.TestContext.NodeName
+	return "localhost"
+}
+
+func metricsURL(port int, path string) string {
```
Comment (SergeyKanzhelev) [inline test/e2e_node/util.go:646]: did you try to unify the logic of creating the URL?

Comment (SergeyKanzhelev) [reply test/e2e_node/util.go:646]: this needs a return

Comment (dims) [reply test/e2e_node/util.go:646]: whoops, didn't mean to push this

**Review by SergeyKanzhelev — APPROVED** [review]
from node perspective might be better to rename the method. But it is just a nit

/lgtm

**Comment by aojea** [comment]
> from node perspective might be better to rename the method. But it is just a nit
> 
> /lgtm

or just revert this https://github.com/kubernetes/kubernetes/commit/a75b00ea39d67a51c6692f9a3338c13306150098 and replace the context by localhost, but is another nit

**Pull request merged**